src/anm/analytics.js

```diff
@@ -0,0 +1,77 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = supportSendBeacon ? 2000 : 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics');
+
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+                setTimeout(event, timeout);
```

Comment:
sorry I didn't get your comments.

I'm not postponing, I'm scheduling the check and if queue is not empty do the request, preferably using `sendBeacon`